src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -3366,9 +3866,9 @@ namespace Microsoft.FSharp.Core
         [<CompiledName("DefaultValueArg")>]
         let defaultValueArg arg defaultValue = match arg with ValueNone -> defaultValue | ValueSome v -> v
 
-        [<NoDynamicInvocation>]
+        [<NoDynamicInvocation(isLegacy=true)>]
```

Comment:
Were we not able to get rid of "[<NoDynamicInvocation>]" for these functions?

Reply:
Working on it

Reply:
@TIHan I looked at this again.  

1. The attribute is still needed to codegen the exception-raising stub body for some non-legacy cases such as the functions in `NatrivePtr.fs`

2. The attribute is needed with Legacy=true for generating the exception-raising stub body for the non-witness-passing entry points for the other cases

The generating code is here:
```
                let attr = TryFindFSharpBoolAttributeAssumeFalse cenv.g cenv.g.attrib_NoDynamicInvocationAttribute v.Attribs
                if (not hasWitnessArgs && attr.IsSome) ||
                   (hasWitnessArgs && attr = Some false) then
                    let exnArg = mkString cenv.g m (FSComp.SR.ilDynamicInvocationNotSupported(v.CompiledName g.CompilerGlobalState))
                    let exnExpr = MakeNotSupportedExnExpr cenv eenv (exnArg, m)
                    mkThrow m returnTy exnExpr
                else
                    body
```

I've removed the attribute from the `prim-types.fsi` and `nativeptr.fsi` signature files since it is not relevant "from the outside", and only relevant to guiding codegen for FSharp.Core.

I think that resolves this point.